cfpb/cfgov-refresh
# TDP: Fix ability to build its assets; add JS unit tests

#5818 brought the Activity Search and Building Blocks tools in from the teachers-digital-platform repository, but wasn't quite complete when it came to building the requisite assets for those tools. This PR should fix that build process.

## Additions

- `package.json` and associated Yarn files for the front-end requirements
- Adds the `tdp-analytics.js` file that was mistakenly not brought along
- Adds the JS unit tests that were not in the original PR

## Removals

- Removes `aos.css` and builds its contents into tdp's `main.less` so we don't have to check in vendor CSS
- Removes copies of JS util modules that are already in cfgov-refresh

## Changes

- Corrects locations of app-specific CSS and JS called in the templates
- Updates tdp's `main.less` to use the Design System packages that come with cfgov-refresh, rather than older Capital Framework packages
- Fixes some accidental removals of `tdp-` in CSS and JS
- Fixes typo in folder name: cfgov/unprocessed/apps/**teachers-digital-plaform** (missing the last `t`) → teachers-digital-platform
- Fixes various lint warnings


## How to test this PR

1. Pull branch
1. `yarn` – confirm that it installs dependencies from the teachers-digital-platform `package.json` file.
1. `gulp clean && gulp build`
1. Visit http://localhost:8000/practitioner-resources/youth-financial-education/journey/ and click through it
1. Visit https://www.consumerfinance.gov/practitioner-resources/youth-financial-education/teach/activities/ and run a search, select some filters, view an activity page, etc.


## Notes

- Documentation purposefully not updated until we complete the transition in the teachers-digital-platform repo
- Removes copies of utility JS modules that were already in cfgov-refresh


## Checklist
